src/syft/core/node/common/client.py

```diff
@@ -383,10 +383,20 @@ def __len__(self) -> int:
 
     def __getitem__(self, key: Union[str, int]) -> Pointer:
         if isinstance(key, str):
+            matches = 0
+            match_obj: Optional[Pointer] = None
             for obj in self.store:
-                if key == str(obj.id_at_location.value):
+                if key in str(obj.id_at_location.value).replace("-", ""):
```

Comment:
Q: Why is that replace needed?

Reply:
In the pandas output we are replacing the - so that you can double click the text and copy and paste it easily. We should refactor all of this to a more robust "search" method which does stuff like partial matching, case insensitive and stripping weird characters etc.